browser/src/app/TilesMiddleware.ts

```diff
@@ -331,24 +406,25 @@ class TileManager {
 				if (!tile.current) this.removeTile(keys[i]);
 			}
 		}
+
```

Comment:
Hmm - I was really trying to avoid any 'real' freeing of canvases during active use. We should 'free' back onto the free list and presumably the code will stop us from allocating new canvases > than that limit (?)

Reply:
I'm not keen on this either, but with the right numbers, this ought to do nothing. Presumably, the right number is likely `highNumCanvases - lowNumCanvases` (the amount of canvases regularly being returned to the pool due to garbage collection), maybe with some safety buffer... Given that `highNumCanvases` is currently double `lowNumCanvases`, this means this is probably about right I think, though a comment explaining this is probably warranted (and explicit testing).

Reply:
Verified this and left a comment in the code - under normal situations (which I also briefly tested), no canvases will be freed. I don't see a path in which cached canvases will get freed during normal use except in some kind of exceptional circumstance where we suddenly allocate a bunch of tiles considerably over the high canvas limit (and in that situation, I think having a failsafe is a good idea).

Reply:
I did also change the values used to explicitly work under this assumption - so now we keep the canvas cache to `highNumCanvases - lowNumCanvases` and we trim it when it reaches above a multiple of that (I arbitrarily picked 1.5, that seemed reasonable).